Refactory packager tiffdss

jeffsuperglide opened the pull request:
packager refactor with tiffdss C function taking GeoTiff data to DSS record.

jeffsuperglide committed:
WIP: packager refactor

jeffsuperglide committed:
WIP: Refactor, add tiffdss code

jeffsuperglide committed:
WIP: added submodule for hec-dss code

jeffsuperglide committed:
WIP: removed submodule hec-dss

jeffsuperglide committed:
WIP: Save point

jeffsuperglide committed:
WIP: Dockerfile in packager supporting tiffdss

jeffsuperglide committed:
WIP: added hec-dss submodule supporting local development

jeffsuperglide committed:
C program for tiff->dss and packager refactor

jeffsuperglide committed:
Updates to C, add docs, test products

jeffsuperglide committed:
Merge branch 'develop' into refactory-packager-tiffdss

jeffsuperglide committed:
Added readme in packager bin for dir placeholder

jeffsuperglide committed:
Corrections to dss7 writer and tiffdss args with spaces

jeffsuperglide committed:
callback updates and GDAL to 3.4.3

jeffsuperglide committed:
Refactory packager tiffdss (#287)

* WIP: packager refactor

* WIP: Refactor, add tiffdss code

* WIP: added submodule for hec-dss code

* WIP: removed submodule hec-dss

* WIP: Save point

* WIP: Dockerfile in packager supporting tiffdss

* WIP: added hec-dss submodule supporting local development

* C program for tiff->dss and packager refactor

* Updates to C, add docs, test products

* Added readme in packager bin for dir placeholder

* Corrections to dss7 writer and tiffdss args with spaces

* callback updates and GDAL to 3.4.3

jeffsuperglide committed:
QUEUE_NAME too QUEUE_NAME_PACKAGER (#289)

jeffsuperglide committed:
Hotfix/http2 (#290)

* Checking http2 and DEBUG logger

* Checking http2 and DEBUG logger

jeffsuperglide committed:
Progress as int (#291)

jeffsuperglide committed:
Merge branch 'develop' of https://github.com/USACE/cumulus-api into develop

jeffsuperglide committed:
WIP ctypes and DSS

jeffsuperglide committed:
Merge branch 'develop' of https://github.com/USACE/cumulus-api into develop

jeffsuperglide committed:
WIP merged from develop

jeffsuperglide committed:
Removed .prj files

jeffsuperglide committed:
Merge branch 'refactory-packager-tiffdss' into feature/packager-ctypes

jeffsuperglide committed:
WIP testing dss7 plugin

jeffsuperglide committed:
WIP: Needs testing

jeffsuperglide committed:
final testing and file cleanup

jeffsuperglide committed:
uncomment appuser

jeffsuperglide committed:
Solved memory leak with multiprocessing and update docs

jeffsuperglide committed:
Solved memory w/ multiprocessing and updated docs

adamscarberry commented on async_packager/Dockerfile:
```diff
@@ -1,26 +1,55 @@
-COPY packager /app/packager
-WORKDIR /app/packager
+# Packager
+FROM osgeo/gdal:ubuntu-small-3.4.3 as packager
```
Might want to bump to 3.5.0

jeffsuperglide replied:
I must have rolled it back at some point but I did have it at 3.5.  There are some dependencies that may need updating too.  Removing from the setup, installing and doing a pip list or freeze would be good to update those requirements.

adamscarberry commented on async_packager/setup.cfg:
```diff
@@ -0,0 +1,39 @@
+    = src
+packages = find:
+python_requires = >=3.6
+install_requires =
```
Are all these modules the latest stable version?

adamscarberry commented on async_packager/setup.cfg:
```diff
@@ -0,0 +1,39 @@
+[metadata]
+name = cumulus-packager
+version = 0.1.0
```
I would call this 2.0

jeffsuperglide replied:
Cumulus-packager is separate from cumulus-geproc but with similarities.  I'd prefer having one package for both containers; less to maintain.

adamscarberry commented on async_packager/setup.cfg:
```diff
@@ -0,0 +1,39 @@
+long_description = file: README.md, LICENSE
+license = MIT
+long_description_content_type = text/markdown
+url = https://docs.rsgis.dev
```
This url looks bogus.  not sure it really matters at this point.

jeffsuperglide replied:
More of a placeholder.  Should have a better url.

adamscarberry reviewed (commented):
Overall looks good.  I made a few comments.

jeffsuperglide committed:
Address comments and button up settings

- Python package `packager` setup and dependencies updated to
stable versions

- Packger Dockerfile updated base image to GDAL 3.5.0

- Packager logging default to `INFO`

- GitHub pages updated to include `packager` code docs
